pkg/client/elasticsearch/query.go

```diff
@@ -114,15 +114,15 @@ func getMetricForPod(
 	defer res.Body.Close()
 
 	if res.IsError() {
-		var e map[string]interface{}
-		if err := json.NewDecoder(res.Body).Decode(&e); err != nil {
+		var errorResponse ErrorResponse
+		if err := json.NewDecoder(res.Body).Decode(&errorResponse); err != nil {
 			return timestampedMetric{}, fmt.Errorf("error parsing the response body: %s", err)
```

Comment:
It would be useful to have the HTTP status and response body in case of an error. I think it's fine to read the body in case of an error instead of just decoding the stream which is a bit more efficient, but we keep it for the main code path when there is no errors.

```go
	if res.IsError() {
		bodyBytes, err := io.ReadAll(res.Body)
		if err != nil {
			return timestampedMetric{}, errors.Wrap(err, "failed to read body")
		}
		var errorResponse ErrorResponse
		if err := json.Unmarshal(bodyBytes, &errorResponse); err != nil {
			return timestampedMetric{}, fmt.Errorf("[%s] failed to get metric pod: %s", res.Status(), string(bodyBytes))
		}
		return timestampedMetric{}, fmt.Errorf("[%s] %s: %s",
			res.Status(),
			errorResponse.Error.Type,
			errorResponse.Error.Reason,
		)
	}

	var r map[string]interface{}
	if err := json.NewDecoder(res.Body).Decode(&r); err != nil {
		return timestampedMetric{}, errors.Wrap(err, "failed to parse response")
	}
```